Sprint 8 solution time and duration

vinnipooha opened the pull request:
Доп. задание не сделала ибо не могу понять, с какого бока к Optional подойти

vinnipooha committed:
Try to fix idea folder

vinnipooha committed:
Try to fix idea folder

vinnipooha committed:
Try to fix idea folder

vinnipooha committed:
TS8: добавить сортировку по приоритету и проверку пересечения по времени

vinnipooha committed:
TS8: Обновить тесты в соответствии с новым функционалом

andrew-larkin commented on src/manager/CSVFormat.java:
```diff
@@ -3,11 +3,14 @@
 
 public class CSVFormat {
-     static Task taskFromString(String value) throws RuntimeException {
+    static Task taskFromString(String value) throws RuntimeException {
```
public

andrew-larkin commented on src/manager/CSVFormat.java:
```diff
@@ -3,11 +3,14 @@
 import java.util.ArrayList;
 import java.util.List;
 
 public class CSVFormat {
```
очень хорошо, что ты вынесла эти методы в отдельный класс!

andrew-larkin commented on src/manager/InMemoryTaskManager.java:
```diff
@@ -23,16 +27,56 @@ public InMemoryTaskManager(HistoryManager historyManager) {
         this.epics = new HashMap<>();
         this.subTasks = new HashMap<>();
         this.historyManager = historyManager;
+        this.prioritizedTasks = new TreeSet<>(Comparator.comparing(Task::getStartTime, Comparator.nullsFirst(naturalOrder())).thenComparing(Task::getId));
```
отлично!

andrew-larkin commented on src/manager/InMemoryTaskManager.java:
```diff
@@ -23,16 +27,56 @@ public InMemoryTaskManager(HistoryManager historyManager) {
+        this.prioritizedTasks = new TreeSet<>(Comparator.comparing(Task::getStartTime, Comparator.nullsFirst(naturalOrder())).thenComparing(Task::getId));
+    }
+
+    public boolean isIntersect(Task task) {
```
private

andrew-larkin commented on src/model/Task.java:
```diff
@@ -9,29 +12,47 @@ public class Task {
     }
 
-    public Task(int id, String name, String description, Status status) {
+//    public Task(String name, String description) {
```
закомментированный код удаляем

andrew-larkin commented on Test/model/EpicTest.java:
```diff
@@ -2,14 +2,23 @@
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
 class EpicTest {
```
большая часть тестов не проходит - необходимо указывать конкретную реализацию TaskManager

andrew-larkin reviewed (changes requested):
Привет! Новый функционал реализован очень даже хорошо, но вот тесты сейчас в не очень хорошей кондиции - если запустить их скопом, почти половина падает, это надо поправить в первую очередь. И небольшие правки по коду.

vinnipooha committed:
TS8: FIX_Исправлены ошибки по ревью_1

vinnipooha committed:
TS8: FIX_Исправлен .gitignore, добавлена папка sourses

vinnipooha committed:
TS8: FIX_Оптимизировано тестирование:
1. Тесты из TaskTest, EpicTest и SubTaskTest вынесены в TaskManagerTest
2. В FileBackedTaskManagerTest реализовано удаление временных файлов после отработки тестов
3. В FileBackedTaskManager в методе main реализовано удаление создаваемого файла из папки sourses
